### ISSUE TITLE
Can't build Terraform due to hcldec related error

### ISSUE THREAD

**Issue by fredsted** [hashicorp/terraform#16686]
I've tried cloning terraform on master, v0.11.0 and v0.10.0 with the same result:

```
% make

==> Checking that code complies with gofmt requirements...
go generate ./...
2017/11/17 13:10:51 Generated command/internal_plugin_list.go
stringer: checking package: decoder_spec.go:4:2: could not import github.com/hashicorp/hcl2/hcldec (/Users/simon/Dev/gocode/pkg/darwin_amd64/github.com/hashicorp/terraform/vendor/github.com/hashicorp/hcl2/hcldec.a: cannot import, possibly version skew (unknown export format version 5 ("version 5")) - reinstall package)
config/configschema/schema.go:80: running "stringer": exit status 1
stringer: checking package: diagnostics.go:4:2: could not import bytes (/usr/local/go/pkg/darwin_amd64/bytes.a: cannot import, possibly version skew (unknown export format version 5 ("version 5")) - reinstall package)
tfdiags/diagnostic.go:11: running "stringer": exit status 1
make: *** [generate] Error 1
```

I've tried:

* installing it with `go get github.com/hashicorp/hcl2/hcldec`
* using `make` OR `make dev` results in the same error

What am I doing wrong?

* Go version 1.9.2
* macOS 10.13.2
* make 3.81

**Comment by jbardin** [comment]
Hi @fredsted,

The error is coming from the `stringer` tool, so it looks like that installation may be broken, or maybe some other part of your go toolchain.

I would start by reinstalling that specific tool with `go get -u golang.org/x/tools/cmd/stringer`

**Comment by nbering** [comment]
Updating `stringer` fixed for me in #16701. Whatever issue there was with stringer didn't affect me as far back is v0.10.0, though. I managed to bisect it directly to a specific commit that caused things to start acting up. Seemed like very strange behaviour, but it's fixed now so it's all good.

**Comment by jbardin** [comment]
@nbering, Thanks for confirming. Those changes were just the first new types to generate stringer output in a while, so your `stringer` tool probably hadn't needed to generate anything since it was built against an older library version.
